crates/services/consensus_module/poa/src/service_test.rs

```diff
@@ -397,3 +425,124 @@ fn test_signing_key() -> Secret<SecretKeyWrapper> {
+        Ok(UncommittedResult::new(
+            ExecutionResult {
+                block: Default::default(),
+                skipped_transactions: Default::default(),
+                tx_status: Default::default(),
+                events: Default::default(),
+            },
+            Default::default(),
+        ))
+    }
+
+    async fn produce_predefined_block(
+        &self,
+        block: &Block,
+    ) -> anyhow::Result<UncommittedResult<Changes>> {
+        self.block_sender
+            .send(FakeProducedBlock::Predefined(block.clone()))
+            .await
+            .unwrap();
+        Ok(UncommittedResult::new(
+            ExecutionResult {
+                block: block.clone(),
+                skipped_transactions: Default::default(),
+                tx_status: Default::default(),
+                events: Default::default(),
+            },
+            Default::default(),
+        ))
+    }
+}
+
+fn block_for_height(height: u32) -> Block {
+    let mut header = PartialBlockHeader::default();
+    header.consensus.height = height.into();
+    let transactions = vec![];
+    Block::new(header, transactions, Default::default(), Default::default()).unwrap()
+}
+
+#[tokio::test]
+async fn consensus_service__run__will_include_predefined_blocks_before_new_blocks() {
+    // given
+    let blocks: [(BlockHeight, Block); 3] = [
+        (2u32.into(), block_for_height(2)),
+        (3u32.into(), block_for_height(3)),
+        (4u32.into(), block_for_height(4)),
+    ];
+    let blocks_map: HashMap<_, _> = blocks.clone().into_iter().collect();
+    let (block_producer, mut block_receiver) = FakeBlockProducer::new();
+    let last_block = BlockHeader::new_block(BlockHeight::from(1u32), Tai64::now());
+    let config = Config {
+        trigger: Trigger::Interval {
+            block_time: Duration::from_millis(100),
+        },
+        signing_key: Some(test_signing_key()),
+        metrics: false,
+        ..Default::default()
+    };
+    let mut block_importer = MockBlockImporter::default();
+    block_importer.expect_commit_result().returning(|_| Ok(()));
+    block_importer
+        .expect_block_stream()
+        .returning(|| Box::pin(tokio_stream::empty()));
+    let mut rng = StdRng::seed_from_u64(0);
+    let tx = make_tx(&mut rng);
+    let TxPoolContext { txpool, .. } = MockTransactionPool::new_with_txs(vec![tx]);
+    let task = MainTask::new(
+        &last_block,
+        config,
+        txpool,
+        block_producer,
+        block_importer,
+        generate_p2p_port(),
+        InMemoryPredefinedBlocks::new(blocks_map),
+    );
+
+    // when
+    let service = ServiceRunner::new(task);
+    service.start().unwrap();
+
+    // then
+    for (_, block) in blocks {
+        let expected = FakeProducedBlock::Predefined(block);
+        let actual = block_receiver.recv().await.unwrap();
+        assert_eq!(expected, actual);
+    }
+    let maybe_produced_block = block_receiver.recv().await.unwrap();
```

Comment:
Could this test also verify the production of the block after pre-defined blocks, please? Like 3 predefined blocks, and 1-3 non predefined(from interval)